private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -33,24 +34,168 @@ export class TemplateServiceClient {
             throw error;
         }
 
-        this.storageProvider = clientOptions.storageProvider;
-        this.authProvider = clientOptions.authenticationProvider;
+        return new TemplateServiceClient(clientOptions.storageProvider, clientOptions.authenticationProvider);
+    }
+    
+    /**
+     * @public
+     * Deletes user info - can only delete own user and all templates created by that user
+     */
+    public async removeUser(): Promise<JSONResponse<Number>> {
+        let owner = this.authProvider.getOwner();
+        if (!owner){
+            return { success: false, errorMessage: ServiceErrorMessage.AuthFailureResponse };
+        }
```

Comment:
This part can go into a method to remove code duplication

Reply:
Done!